src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -252,6 +252,18 @@ mul_nonsing_inv A (is_unit_det_of_invertible A)
 @[simp] lemma inv_mul_of_invertible [invertible A] : A⁻¹ ⬝ A = 1 :=
 nonsing_inv_mul A (is_unit_det_of_invertible A)
 
+@[simp] lemma mul_inv_cancel_right (B : matrix m n α) [invertible A] : B ⬝ A ⬝ A⁻¹ = B :=
+by simp [matrix.mul_assoc]
+
+@[simp] lemma mul_inv_cancel_left (B : matrix n m α) [invertible A] : A ⬝ (A⁻¹ ⬝ B) = B :=
+by simp [←matrix.mul_assoc]
+
+@[simp] lemma inv_mul_cancel_right (B : matrix m n α) [invertible A] : B ⬝ A⁻¹ ⬝ A = B :=
+by simp [matrix.mul_assoc]
+
+@[simp] lemma inv_mul_cancel_left (B : matrix n m α) [invertible A] : A⁻¹ ⬝ (A ⬝ B) = B :=
+by simp [←matrix.mul_assoc]
```

Comment:
I think these should all have `of_invertible` in their names, and be proved in terms of a `nonsing_inv` lemma

bors r-

bors d+

Reply:
I hope I understood you correctly. Please have a look at my commit [485d7ee](https://github.com/leanprover-community/mathlib/pull/14644/commits/485d7eef8a753503d9066bf49740def4c31aa16f).